x/auth/client/cli/tx_sign.go

```diff
@@ -63,12 +63,17 @@ account key. It implies --signature-only.
 func makeSignBatchCmd(cdc *codec.Codec) func(cmd *cobra.Command, args []string) error {
 	return func(cmd *cobra.Command, args []string) error {
 		inBuf := bufio.NewReader(cmd.InOrStdin())
-		clientCtx := client.NewContextWithInput(inBuf).WithCodec(cdc)
-		txBldr := types.NewTxBuilderFromCLI(inBuf)
+		clientCtx := client.Context{}.WithInput(inBuf).WithCodec(cdc)
```

Comment:
Any reason not to use `clientCtx := client.GetClientContextFromCmd(cmd)` here? You'll get `home` populated, so then you just need to do `clientCtx.HomeDir` instead of reading the flag.